src/containers/ProfileDetails/components/CaptivePortal/index.js

```diff
@@ -260,6 +254,17 @@ const CaptivePortalForm = ({ details, form, fileUpload }) => {
 
   const handleDeleteWhitelist = item => setWhitelist(whitelist.filter(i => i !== item));
 
+  const handleAddUser = newUser => {
+    details.userList.push(newUser);
+  };
+  const handleUpdateUser = (index, newUser) => {
+    details.userList.splice(index, 1, newUser);
+  };
+
+  const handleDeleteUser = index => {
+    details.userList.splice(index);
+  };
+
```

Comment:
i already told you never modify props. the userList should be implemented the same way as whitelist